parsec/bindthread.c

```diff
@@ -19,7 +19,8 @@
 #elif defined(PARSEC_HAVE_SCHED_SETAFFINITY)
 #  include <linux/unistd.h>
 #  include <sched.h>
-#elif defined(PARSEC_OSX)
+#endif  /* define(PARSEC_HAVE_HWLOC) */
+#if defined(__APPLE__)
```

Comment:
What happens if hwloc starts supporting OS X? We'll have both the OS X-specific includes / code paths below and the hwloc code. In other words, why replace the `elif` with `endif`, `if`?

Reply:
We can compile with hwloc today (that's actually how I run it), all features are supported, that is we can work on cpusets etc to detect problematic binding conditions, but the bind operation itself returns an error and doesn't do anything. 

That's why we have a specific code path to use the OS specific binding operation (it's not quite a real bind, but an approximation of it that will mark a preference for the target cpu, and an aversion for migration, but that's the best that can be done on OSX).